test/controllers/api/v2/puppet_modules_controller_test.rb

```diff
@@ -34,11 +34,41 @@ def test_index_by_env
       assert_template "katello/api/v2/puppet_modules/index"
     end
 
+    test_attributes :pid => 'eafc7a71-d550-4983-9941-b87aa57b83e9'
+    def test_index_by_repo_empty_results
+      repo = katello_repositories(:dev_p_forge)
+      get :index, params: { :repository_id => repo.id }
+      assert_response :success
+      assert_template "katello/api/v2/puppet_modules/index"
+      results = JSON.parse(response.body)
+      assert results.key?('results')
+      assert_empty results['results']
+    end
+
+    test_attributes :pid => '5337b2be-e207-4580-8407-19b88cb40403'
+    def test_index_by_repo_single_result
+      repo = katello_repositories(:dev_p_forge)
+      repo_puppet_module = RepositoryPuppetModule.new(:repository => repo, :puppet_module => @puppet_module)
+      repo_puppet_module.save!
+      get :index, params: { :repository_id => repo.id }
+      results = JSON.parse(response.body)
+      assert results.key?('results')
```

Comment:
there are a few `assert results.key?('results')` in your change. I think they're not necessary because you go on to read from the key you are checking which means you are implicitly testing for that key's existence already